### PR TITLE
Added code to reset ukf filter

### PR DESCRIPTION
**Pull request by costashatz**
Added code to reset ukf filter (robot_localization node). This code should work even if no robot_localization node is running, as we are just publishing some data on a topic.

### CODE REVIEW

File: src/robotHexa.cpp [hunk 1]
```diff
@@ -681,6 +681,9 @@ void RobotHexa :: initRosNode(  int argc ,char** argv,boost::shared_ptr<ros::Nod
   _node_p = node_p;
   //    _chatter_pub  = _node_p->advertise<rgbdslam::SferesCmd>("sferes_cmd", 1);
 
+  // subscribe to filter reset
```
Comment (dogoepp) [inline src/robotHexa.cpp]: What is filter reset  meaning ?
Do we subscribe or advertise ?

File: src/robotHexa.cpp [hunk 2]
```diff
@@ -760,6 +763,23 @@ void RobotHexa :: transfer(ControllerDuty& controller, float duration,int transf
     ROS_INFO_STREAM("Failed to reset odometry");
   }
 
+  // reset filter
```
Comment (dogoepp) [inline src/robotHexa.cpp]: Maybe something like "send a message to reset the sensor fusion filter used for odometry, if there is any" could be a little better ?

File: src/robotHexa.cpp [hunk 3]
```diff
@@ -760,6 +763,23 @@ void RobotHexa :: transfer(ControllerDuty& controller, float duration,int transf
...
+  tmp.pose.pose.position.y = 0;
+  tmp.pose.pose.position.z = 0;
+
+  tmp.pose.pose.orientation.x = 0;
+  tmp.pose.pose.orientation.y = 0;
+  tmp.pose.pose.orientation.z = 0;
+  tmp.pose.pose.orientation.w = 1;
+
+  _reset_filter.publish(tmp);
+  ROS_INFO_STREAM("reset_filter sent");
```
Comment (dogoepp) [inline src/robotHexa.cpp]: Could this message be a little less elusive ? My weak memory would soon forget what this means ;-)

**Comment by dogoepp** [comment]
I hope the tone of the comments is not offending :$

**Comment by costashatz** [comment]
@dogoepp Haha.. Not at all! It was a quick PR so not to forget to do it. I am fixing the issues. I will create some more PRs this week to clean this repo a little.

**Comment by costashatz** [comment]
Should be more clear now!

**Comment by dogoepp** [comment]
**OK** for me

**Comment by jbmouret** [comment]
What happens if the IMU is not detected?

**Comment by costashatz** [comment]
This PR does not deal with that. There are 2 usages of the experiment (until now):
- Use only rgdb visual odometry. 
  - In this case we subscribe only to rtabmap's node for the visual odometry to get the position. This is done in `resibots_launch`.
  - The code above works fine, because resetting the UKF filter (which in this case does not exist) is done just by publishing to a topic. So, we just publish to a topic no one reads.
- Use rgbd visual odometry and IMU reading combined with a UKF filter (using robot_localization package).
  - In this case we subscribe to UKF filter's output to get the position. If the IMU is not detected, the filter will rely only on the rgbd sensor. As far as I can understand from the documentation. This is done in `resibots_launch`. Certainly, it's better to use the visual odometry data directly if we have no IMU. But this should be handled in `resibots_launch`, not here.
  - The code above again works fine as it resets the actual filter.

I am preparing more PRs to clean the code and pass all possible configurations from the parameter server and then everything will stick properly into place.

**Comment by jbmouret** [comment]
Anyway, **OK for me**

**Pull request merged**